plugins/rocketmq/src/main/java/com/navercorp/pinpoint/plugin/rocketmq/interceptor/ProducerSendCallBackInterceptor.java

```diff
@@ -0,0 +1,109 @@
+
+package com.navercorp.pinpoint.plugin.rocketmq.interceptor;
+
+import org.apache.rocketmq.client.producer.SendResult;
+import org.apache.rocketmq.client.producer.SendStatus;
+
+import com.navercorp.pinpoint.bootstrap.async.AsyncContextAccessor;
+import com.navercorp.pinpoint.bootstrap.context.AsyncContext;
+import com.navercorp.pinpoint.bootstrap.context.MethodDescriptor;
+import com.navercorp.pinpoint.bootstrap.context.SpanEventRecorder;
+import com.navercorp.pinpoint.bootstrap.context.TraceContext;
+import com.navercorp.pinpoint.bootstrap.interceptor.AsyncContextSpanEventSimpleAroundInterceptor;
+import com.navercorp.pinpoint.bootstrap.interceptor.SpanEventSimpleAroundInterceptorForPlugin;
+import com.navercorp.pinpoint.plugin.rocketmq.RocketMQConstants;
+
+/**
+ * @author messi-gao
+ */
+public final class ProducerSendCallBackInterceptor {
+    public static class ConstructInterceptor extends SpanEventSimpleAroundInterceptorForPlugin {
+        public ConstructInterceptor(TraceContext traceContext, MethodDescriptor descriptor) {
+            super(traceContext, descriptor);
+        }
+
+        @Override
+        protected void doInBeforeTrace(SpanEventRecorder recorder, Object target, Object[] args) {
+            final AsyncContext asyncContext = recorder.recordNextAsyncContext();
+            ((AsyncContextAccessor) target)._$PINPOINT$_setAsyncContext(asyncContext);
+        }
+
+        @Override
+        protected void doInAfterTrace(SpanEventRecorder recorder, Object target, Object[] args, Object result,
+                                      Throwable throwable) {
+            recorder.recordServiceType(RocketMQConstants.ROCKETMQ_CLIENT_INTERNAL);
+            recorder.recordApi(methodDescriptor);
+            recorder.recordException(throwable);
+        }
+    }
+
+    public static class OnSuccessInterceptor extends AsyncContextSpanEventSimpleAroundInterceptor {
+        public OnSuccessInterceptor(
+                TraceContext traceContext, MethodDescriptor methodDescriptor) {
+            super(traceContext, methodDescriptor);
+        }
+
+        @Override
+        protected void doInBeforeTrace(SpanEventRecorder spanEventRecorder, AsyncContext asyncContext, Object o,
+                                       Object[] objects) {
+        }
+
+        @Override
+        protected void doInAfterTrace(SpanEventRecorder spanEventRecorder, Object target, Object[] args,
+                                      Object result, Throwable throwable) {
+            spanEventRecorder.recordServiceType(RocketMQConstants.ROCKETMQ_CLIENT_INTERNAL);
+            spanEventRecorder.recordApi(methodDescriptor);
+            final Object arg = args[0];
+            if (arg instanceof SendResult) {
+                final SendStatus sendStatus = ((SendResult) arg).getSendStatus();
+                spanEventRecorder.recordAttribute(RocketMQConstants.ROCKETMQ_SEND_STATUS_ANNOTATION_KEY,
+                                                  sendStatus);
```

Comment:
It is better to use sendStatus.name() instead of sendStatus.

Reply:
fix it